internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go

```diff
@@ -0,0 +1,141 @@
+package builder
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/consul/envoyextensions/xdscommon"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1/pbproxystate"
+)
+
+func (b *Builder) BuildLocalApp(workload *pbcatalog.Workload) *Builder {
+	return b.addInboundListener(xdscommon.PublicListenerName, workload).
+		addInboundRouters(workload).
+		addInboundTLS()
+}
+
+func (b *Builder) getLastBuiltListener() *pbproxystate.Listener {
+	lastBuiltIndex := len(b.proxyStateTemplate.ProxyState.Listeners) - 1
+	return b.proxyStateTemplate.ProxyState.Listeners[lastBuiltIndex]
+}
+
+func (b *Builder) addInboundListener(name string, workload *pbcatalog.Workload) *Builder {
+	listener := &pbproxystate.Listener{
+		Name:      name,
+		Direction: pbproxystate.Direction_DIRECTION_INBOUND,
+	}
+
+	// We will take listener bind port from the workload for now.
```

Comment:
Should this have a `TODO` to track that it sounds this is temporary?

Reply:
I think this comment is wrong and I updated it. Originally, I was looking at our current implementation where we allow you to override this address with a `BindAddress` in opaque config. But in the new model, you can just specify a different address for your mesh port instead and so we don't need to look in other places.